pkg/cache/v2/cache.go

```diff
@@ -72,6 +87,21 @@ type Response interface {
 	GetVersion() (string, error)
 }
 
+// DeltaResponse is a wrapper around Envoy's DeltaDiscoveryResponse
+type DeltaResponse interface {
+	// Get the constructed DeltaDiscoveryResponse
+	GetDeltaDiscoveryResponse() (*discovery.DeltaDiscoveryResponse, error)
+
+	// Get te original Request for the Response.
+	GetDeltaRequest() *discovery.DeltaDiscoveryRequest
+
+	// Get the version in the Response.
+	GetSystemVersion() (string, error)
+
+	// Get the version map of the internal cache
+	GetDeltaVersionMap() map[string]string
```

Comment:
This is here because we bubble up this generated version map into the server for it to use as a state tracker. For STOW it's similar to how `VersionInfo` was used, so `GetVersion()` equates to `GetVersionMap()`. They're used differently between the protocols but we still needed a way to pull version information out of the response in the server code since Incremental didn't have a mechanism like that (with the exception of the `system_version_info` debugging field but that is not a reliable source of version information and doesn't account for per resource version changes)

Reply:
Yeah makes sense, especially after I looked back at the other PR how this is used.

How about we try to name it something that makes it a bit clearer that it's essentially the updated version mappings after this response is applied? 

As I understand it for a given delta stream we have version map at time T that we pass to CreateWatch, and it returns the version map at T+1, containing the resource versions after we apply the response. So maybe something like `NextVersionMap`, `UpdatedVersionMap`, etc?

I think the key difference between this and the SotW equivalent is that for SotW the VersionInfo doesn't depend on the input and captures the full version info of the response, which makes it a bit easier to grok.

Reply:
Ah yes I like this. I think semantically `NextVersionMap` is my favorite. 

Right that was my understanding as well 👍